tests/cpp/test_multidevice_preseg_passes.cpp

```diff
@@ -0,0 +1,186 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <fusion.h>
+#include <ops/all_ops.h>
+#include <preseg_passes/mark_aliases_prepare.h>
+#include <preseg_passes/optimization_pass.h>
+#include <preseg_passes/propagate_shardings.h>
+#include <runtime/fusion_executor_cache.h>
+#include <tests/cpp/multidevice.h>
+#include <tests/cpp/validator.h>
+#include "multidevice/utils.h"
+
+namespace nvfuser {
+
+constexpr int64_t b = 2, s = 3, h = 16, a = 2;
+constexpr double dropout_p = 0.0;
+constexpr bool is_causal = false;
+
+using testing::ElementsAre;
+using MultiDevicePresegPassesTest = MultiDeviceTest;
+
+TEST_F(MultiDevicePresegPassesTest, ResidualAdd) {
```

Comment:
As a convention, test_multidevice_*.cpp are compiled into test_multidevice, which is executed under mpirun. I don't think this test in particular needs multi-GPU execution. I'd move it to test_sharding.cpp instead.